bin/juttle

```diff
@@ -214,6 +217,10 @@ function perform_run(options) {
             program.deactivate();
         }).finally(function() {
             current_program = undefined;
+
+            if (had_an_error) {
+                return Promise.reject();
```

Comment:
I think in the specific case of the http adapter the status code, message, and url being fetched are sufficient.